src/nrniv/nrncore_write/io/nrncore_io.cpp

```diff
@@ -390,25 +397,84 @@ void write_nrnthread_task(const char* path, CellGroup* cgs)
+        //   Note: The number of groupids (2nd or 3rd line) has to be
+        //   overwritten wih the total number so far. To avoid copying
+        //   old to new, we allocate 10 chars for that number.
+
         std::stringstream ss;
         ss << path << "/files.dat";
 
         std::string filename = ss.str();
 
-        FILE *fp = fopen(filename.c_str(), "w");
-        if (!fp) {
-            hoc_execerror("nrnbbcore_write write_nrnthread_task could not open for writing:", filename.c_str());
+        FILE *fp = NULL;
+        if (append == false) { // start a new file
+            fp = fopen(filename.c_str(), "w");
+            if (!fp) {
+                hoc_execerror("nrncore_write write_nrnthread_task could not open for writing:", filename.c_str());
+            }
+        } else { // modify groupid number and append to existing file
+            fp = fopen(filename.c_str(), "r+");
+            if (!fp) {
+                hoc_execerror("nrncore_write write_nrnthread_task could not open for modifying:",  filename.c_str());
+            }
+        } 
+
+        int maxlen = 20;
+        char line[maxlen]; // All lines are actually no larger than %10d.
+
+        if (append) {
+            // verify same version
+            fgets_err(line, maxlen, fp);
+            // unfortunately line has the newline
+            size_t n = strlen(bbcore_write_version);
+            if ((strlen(line) - 1 != n)
+                || strncmp(line, bbcore_write_version, n) != 0) {
+                fclose(fp);
+                hoc_execerror("nrncore_write bad existing files.dat version:", line);
+            }
+        } else {
+            fprintf(fp, "%s\n", bbcore_write_version);
         }
 
-        fprintf(fp, "%s\n", bbcore_write_version);
-
         // notify coreneuron that this model involves gap junctions
         if (nrnthread_v_transfer_) {
-            fprintf(fp, "-1\n");
+            if (append) {
+                fgets_err(line, maxlen, fp);
+                if (strcmp(line, "-1\n") != 0) {
+                    fclose(fp);
+                    hoc_execerror("nrncore_write existing files.dat does not have a gap junction indicator\n", NULL);
+                }
+            } else {
+                fprintf(fp, "-1\n");
+            }
         }
 
         // total number of datasets
-        fprintf(fp, "%d\n", iSumThread);
+        if (append) {
+            // this is the one that needs the space to get a new value
+            long pos = ftell(fp);
+            fgets_err(line, maxlen, fp);
+            int oldval = 0;
+            int i = sscanf(line, "%d\n", &oldval);
+            if (i != 1) {
+                fclose(fp);
+                hoc_execerror("nrncore_write error reading number of groupids", NULL);
+            }
+            if (oldval == -1) {
+                fclose(fp);
+                hoc_execerror("nrncore-write existing files.dat has gap junction indicator where we expected a groupgid count.", NULL);
+            }
```

Comment:
same for above two hoc_execerror : would be better to give hint to user what to do next?

Reply:
Will revise the error messages

Reply:
I think the key info in the error messages is whether nrncore_write is in append mode.